server/store/subscriptions.go

```diff
@@ -58,7 +58,7 @@ func (subscriptionList *SubscriptionList) AddSubscription(userID string, subscri
 		subscriptionList.ByMattermostUserID[userID] = make(SubscriptionListMap)
 	}
 
-	subscriptionKey := GetSubscriptionKey(userID, subscription.ProjectName, subscription.ChannelID, subscription.EventType, subscription.Repository, subscription.TargetBranch, subscription.PullRequestCreatedBy, subscription.PullRequestReviewersContains, subscription.PushedBy, subscription.MergeResult, subscription.NotificationType, subscription.AreaPath)
+	subscriptionKey := GetSubscriptionKey(userID, subscription.ProjectName, subscription.ChannelID, subscription.EventType, subscription.Repository, subscription.TargetBranch, subscription.PullRequestCreatedBy, subscription.PullRequestReviewersContains, subscription.PushedBy, subscription.MergeResult, subscription.NotificationType, subscription.AreaPath, subscription.ReleasePipeline, subscription.BuildPipeline, subscription.BuildStatus, subscription.ApprovalType, subscription.ApprovalStatus, subscription.StageName, subscription.ReleaseStatus)
```

Comment:
do we really need all these in the subscription key?

Reply:
Yes, so that we can differentiate each subscription based on filters